File: ios/Cookpit/Cookpit/MapViewModel.swift
//
//  MapViewModel.swift
//  Cookpit
//
//  Created by Kittinun Vantasin on 7/17/16.
//  Copyright © 2016 Cookpit. All rights reserved.
//

import Foundation
import Mapbox

enum MapViewModelCommand {
  
  case SetItems(items: [CPMapDetailViewData])
  
}

struct MapViewModel {

  let items: [CPMapDetailViewData]
  
  func executeCommand(command: MapViewModelCommand) -> MapViewModel {
    switch command {
    case let .SetItems(items):
      return MapViewModel(items: items)
    }
  }
  
}

extension CPMapDetailViewData : MGLAnnotation {

  public var coordinate: CLLocationCoordinate2D {
    return location.coordinate2D()
  }
  
  public var title: String? {
    return text
  }
  
}

File: android/Cookpit/app/src/main/kotlin/com/github/kittinunf/cookpit/map/MapViewModel.kt
package com.github.kittinunf.cookpit.map

import com.github.kittinunf.cookpit.MapDetailViewData
import com.mapbox.mapboxsdk.annotations.MarkerOptions

sealed class MapViewModelCommand {

    class SetItems(val items: List<MapDetailViewData>) : MapViewModelCommand()

}

class MapViewModel(val items: List<MapDetailViewData> = listOf()) {

    fun executeCommand(command: MapViewModelCommand): MapViewModel {
        when (command) {
            is MapViewModelCommand.SetItems -> {
                return MapViewModel(command.items)
            }
        }
    }

}

fun MapDetailViewData.createMarkerOptions(): MarkerOptions {
    return MarkerOptions().title(text).position(location)
}

 
